### DISCUSSION TITLE
[Run issue]: Problem using Cmake at step 8.4 on WSL

### DISCUSSION THREAD

**Discussion by ShahinGHD** [OpenTOPAS/OpenTOPAS#17] [Run-Problem] [answered]
### Contact Details

shahin.ghasemi@gmail.com

### What was the issue?

Hi

I did all the steps on windows WSL ubuntu 20.04 until step 8.4
everything goes well but:

    export Geant4_DIR=$HOME/Applications/GEANT4/geant4-install
    export GDCM_DIR=$HOME/Applications/GDCM/gdcm-install
    cmake ../OpenTOPAS -DCMAKE_INSTALL_PREFIX=../OpenTOPAS-install

after the third command I get this error:

-- The C compiler identification is GNU 9.4.0
-- The CXX compiler identification is GNU 9.4.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- CMAKE_CXX_COMPILER_ID: GNU
-- CMAKE_CXX_COMPILER_VERSION: 9.4.0
CMake Warning at cmake/Modules/TOPAS_Options.cmake:9 (message):
  Setting CMAKE_BUILD_TYPE=Release
Call Stack (most recent call first):
  CMakeLists.txt:21 (include)

CMake Error at cmake/Modules/TOPAS_FindPackages.cmake:9 (find_package):
  By not providing "FindGeant4.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "Geant4", but
  CMake did not find one.

  Could not find a package configuration file provided by "Geant4" with any
  of the following names:

    Geant4Config.cmake
    geant4-config.cmake

  Add the installation prefix of "Geant4" to CMAKE_PREFIX_PATH or set
  "Geant4_DIR" to a directory containing one of the above files.  If "Geant4"
  provides a separate development package or SDK, be sure it has been
  installed.
Call Stack (most recent call first):
  CMakeLists.txt:22 (include)

-- Configuring incomplete, errors occurred!
See also "/home/shahin/TOPAS/OpenTOPAS-build/CMakeFiles/CMakeOutput.log".

I checked the location of the folders, those are fine, inside geant4-install folder I have :
bin. include, lib and share

Inside the folder GDCM I have:
include, lib and share

I appreciate any help,

Thanks
Shahin

### TOPAS category

Geometry

### Version

4.0

### What Operative System (OS) are you running on?

Windows (WSL)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Attestation

- [X] I testify that I have fully read the bug reporting guideline beforehand.

**Comment by JNaokiDKondo** [comment]
Hi Shahin,
Would please let us know what is stored in your `Geant4_DIR` and `GDCM_DIR` environment variables? This can be done using the following commands:

`echo $Geant4_DIR`
`echo $GDCM_DIR`
`ls -l $Geant4_DIR`
`ls -l $GDCM_DIR`

Basically, the issue seems to be that those environment variables are pointing to the wrong folders. If any of the last two commands yields no output in your console (or returns an error) that would indicate such thing.

-Naoki

**Reply by ShahinGHD** [reply]
It seems it is pointing at a right direction and the ls is also ok

shahin@tinglee11:/TOPAS/OpenTOPAS-build$ export Geant4_DIR=$HOME/GEANT4/geant4-install
shahin@tinglee11:/TOPAS/OpenTOPAS-build$ export GDCM_DIR=$HOME/GDCM/gdcm-install

shahin@tinglee11:/TOPAS/OpenTOPAS-build$ echo $GDCM_DIR
/home/shahin/GDCM/gdcm-install
shahin@tinglee11:/TOPAS/OpenTOPAS-build$ echo $Geant4_DIR
/home/shahin/GEANT4/geant4-install
shahin@tinglee11:~/TOPAS/OpenTOPAS-build$

shahin@tinglee11:/TOPAS/OpenTOPAS-build$ ls -l $Geant4_DIR
total 16
drwxr-xr-x 2 root root 4096 May 6 18:12 bin
drwxr-xr-x 3 root root 4096 May 6 18:12 include
drwxr-xr-x 5 root root 4096 May 6 18:12 lib
drwxr-xr-x 3 root root 4096 May 6 18:12 share
shahin@tinglee11:/TOPAS/OpenTOPAS-build$ ls -l $GDCM_DIR
total 12
drwxr-xr-x 3 root root 4096 May 6 18:04 include
drwxr-xr-x 3 root root 4096 May 6 18:04 lib
drwxr-xr-x 3 root root 4096 May 6 18:04 share

The error still there tho:
-- CMAKE_CXX_COMPILER_ID: GNU
-- CMAKE_CXX_COMPILER_VERSION: 9.4.0
CMake Warning at cmake/Modules/TOPAS_Options.cmake:9 (message):
Setting CMAKE_BUILD_TYPE=Release
Call Stack (most recent call first):
CMakeLists.txt:21 (include)

CMake Error at cmake/Modules/TOPAS_FindPackages.cmake:9 (find_package):
By not providing "FindGeant4.cmake" in CMAKE_MODULE_PATH this project has
asked CMake to find a package configuration file provided by "Geant4", but
CMake did not find one.

Could not find a package configuration file provided by "Geant4" with any
of the following names:

Geant4Config.cmake
geant4-config.cmake
Add the installation prefix of "Geant4" to CMAKE_PREFIX_PATH or set
"Geant4_DIR" to a directory containing one of the above files. If "Geant4"
provides a separate development package or SDK, be sure it has been
installed.
Call Stack (most recent call first):
CMakeLists.txt:22 (include)

-- Configuring incomplete, errors occurred!

**Comment by ShahinGHD** [comment, marked as answer]
I can confirm that removing ubuntu 20.04 and installing ubuntu 24 fixed the issue

**Comment by JNaokiDKondo** [comment]
A quick update on this subject. We now have a QuickStart Guide for WSL in the OpenTOPAS ReadTheDocs found at https://opentopas.readthedocs.io/en/latest/getting-started/WSL.html
